XNAControls/XNAControl.cs

```diff
@@ -372,38 +352,24 @@ public virtual float Alpha
     /// </summary>
     public bool InputEnabled { get; set; } = true;
 
-    private bool isActive = false;
+    private bool isActive;
 
     /// <summary>
     /// Gets or sets a bool that determines whether this control is the current focus of the mouse cursor.
     /// </summary>
     public bool IsActive
     {
-        get
-        {
-            if (Parent != null && !Detached)
-                return Parent.IsActive && isActive;
+        get => Parent != null && !Detached ? Parent.IsActive && isActive : isActive;
 
-            return isActive;
-        }
-        set { isActive = value; }
+        set => isActive = value;
     }
 
-    private bool _ignoreInputOnFrame = false;
+    private bool ignoreInputOnFrame;
 
     public bool IgnoreInputOnFrame
     {
-        get
-        {
-            if (Parent == null)
-                return _ignoreInputOnFrame;
-            else
-                return _ignoreInputOnFrame || Parent.IgnoreInputOnFrame;
-        }
-        set
-        {
-            _ignoreInputOnFrame = true;
-        }
+        get => Parent == null ? ignoreInputOnFrame : ignoreInputOnFrame || Parent.IgnoreInputOnFrame;
```

Comment:
While at it, you could change this to use `AppliesToSelfAndAllParents` instead of only checking its own parent.

Reply:
Not sure what the desired effect is.
You suggest like this?
`get => ignoreInputOnFrame || AppliesToSelfAndAllParents(q => q.IgnoreInputOnFrame);`
Or should we have something like this:
`get => AppliesToSelfOrAnyParent(q => q.IgnoreInputOnFrame);`

Reply:
Right. Yeah, `AppliesToSelfOrAnyParent` would be the desired effect here.